ZenPacks/zenoss/CalculatedPerformance/datasources/AggregatingDataSource.py

```diff
@@ -66,6 +68,32 @@ def getDescription(self):
             (self.targetDataSource, self.targetDataPoint, self.targetRRA, self.targetMethod)
         return description
 
+    def getCycleTime(self, context):
+        """Return collection interval for given context."""
+        if self.useBasisInterval:
+            cycletime = self.getTargetCycleTime(context)
+            if cycletime is not None:
+                return operations._bound(
+                    minValue=self.minimumInterval,
+                    value=cycletime,
+                    maxValue=self.maximumInterval)
+
```

Comment:
Can you comment here why the cycletime must be within the proper bounds? Its not clear from the context.